Sources/PackageLoading/TargetSourcesBuilder.swift

```diff
@@ -621,6 +636,12 @@ public struct FileRuleDescription {
         coredata,
         metal,
     ]
+
+    /// List of file types that are ignored in this version of SwiftPM CLI
+    /// (that don't generate warnings for being unhandled).
+    public static let ignoredFileTypes: [FileRuleDescription] = [
+        docc,
+    ]
```

Comment:
this seems fine for now, but in the future we would need allow plugins to define these

Reply:
another question is if there are others that we need to add here, for example one common example is readme files

Reply:
not sure if we can know for sure whether a README file should be ignored or if it is possibly something that should be a resource

Reply:
Absolutely, this is a minimal change intended to be safe enough to nominate to 5.5 after it's been accepted in main.  As part of the plugin work I think there will need to be some refactoring of the logic here to hold off on emitting warnings until we know the paths that plugins supported (unless we want to extend build tool plugins to allow them to declare suffixes that should be exempt from warnings because the plugin handles them, which would be simpler but would require that extra declaration by the plugin).

Reply:
right, I dont think we need to make any changes to plugin API right now, but we should make sure this change is architecturally aligned / does not paint us to a corner as we move this kind of functionality into plugin

Reply:
Agreed.  Since I think that we will want to have a set of types that are not processed but also not warned about (i.e. ignored), I think that this change is in line with that, even if the set of types will eventually be dynamically extendable by plugins.